Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/SurfaceMagnetism.cs

```diff
@@ -7,96 +7,253 @@
 namespace Microsoft.MixedReality.Toolkit.Utilities.Solvers
 {
     /// <summary>
-    /// SurfaceMagnetism casts rays to Surfaces in the world align the object to the surface.
+    /// SurfaceMagnetism casts rays to Surfaces in the world and aligns the object to the hit surface.
     /// </summary>
     public class SurfaceMagnetism : Solver
     {
-        private const float MaxDot = 0.97f;
-
-        private enum RaycastDirectionEnum
+        #region Enums
+        public enum RaycastDirectionType
         {
-            CameraFacing,
+            CameraFacing = 0,
             ToObject,
             ToLinkedPosition
         }
 
-        private enum OrientModeEnum
+        public enum OrientionMode
         {
-            None,
+            None = 0,
             Vertical,
             Full,
             Blended
         }
+        #endregion
 
+        #region SurfaceMagnetism Parameters
         [SerializeField]
-        [Tooltip("LayerMask to apply Surface Magnetism to")]
-        private LayerMask[] magneticSurfaces = { UnityEngine.Physics.DefaultRaycastLayers };
+        [Tooltip("Array of LayerMask to execute from highest to lowest priority. First layermask to provide a raycast hit will be used by component")]
+        protected LayerMask[] magneticSurfaces = { UnityEngine.Physics.DefaultRaycastLayers };
```

Comment:
I think in other places we'll usually still keep these private, so that:

1) In the future we can change the way we store these values (i.e. the way they are serialized) without it being a breaking change to sub-classes.
2) We already have an external exposure (both to subclasses and consumers) though the public getters/setters, so folks should be able to get the data they need.

Or, is this being done because if it's private, if we subclass it, the subclass's inspectors won't show those private parameters?

Reply:
I can make them all private. I had them protected so that they could be change. The public get/set are there but they could be extended to do other functionality other than just get/set so I thought if someone needed a workaround to change a value without calling the public interfaces that made do other stuff

Reply:
I'm fine with making them private. I made them protected so that if sub-classes want to set the variables they can without activating the setter which might activate other functions not desired by the subclass